src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -23,7 +24,7 @@ public partial class Socket : IDisposable
         internal const int DefaultCloseTimeout = -1; // NOTE: changing this default is a breaking change.
 
         private static readonly IPAddress s_IPAddressAnyMapToIPv6 = IPAddress.Any.MapToIPv6();
-
+        private static readonly byte[] s_zero = new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
```suggestion
        private static ReadOnlySpan<byte> ZeroBytes => new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```
Then just slice that span in https://github.com/dotnet/runtime/blob/98bf687176800b6f8dde0e71deb2ade3d8c90590/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L1376 and https://github.com/dotnet/runtime/blob/98bf687176800b6f8dde0e71deb2ade3d8c90590/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L3028